src/thick-discs.jl

```diff
@@ -0,0 +1,260 @@
+# fitting thick discs with lamp post model
+# MCG--6-30-15 one XMM dataset only for illustrative purposes
+# based on the spin.jl fits
+
+using SpectralFitting
+using Plots
+using Gradus
+using GradusSpectralModels
+using JLD2
+# for JLD2 compression
+using CodecZlib
+# using Reflionx
+using CSV
+using DataFrames
+
+include("additional-models.jl")
+
+function read_or_make_transfer_table(path; new=false)
+    if isfile(path) && !new
+        @info "Reading transfer function table from file..."
+        f = jldopen(path, "r")
+        table = f["table"]
+        close(f)
+        table
+    else
+        @info "Creating new transfer function table..."
+        d = ThinDisc(0.0, Inf)
+        as = collect(range(0.0, 0.998, 10))
+        # 89 fails at the moment, but don't really need full parameter space for
+        # MCG-6 anyway
+        thetas = collect(range(1.0, 88.0, 10))
+        # this takes about 300 seconds or so; using JDL2 if it exists see
+        # https://github.com/astro-group-bristol/Gradus.jl/issues/194
+        table = @time make_transfer_function_table(KerrMetric, d, as, thetas)
+        jldopen(path, "w"; compress=true) do f
+            f["table"] = table
+        end
+        table
+    end
+end
+
+data_path = "data"
+# TRANSFER_FUNCTION_PATH = joinpath(data_path, "thin-disc-transfer-table-900.jld2")
+TRANSFER_FUNCTION_PATH = joinpath(data_path, "thick-disc-transfer-table-new.jld2")
+
+LAMP_POST_PROFILE_PATH = joinpath(data_path, "lamp-post-profile.jld2")
+
+table = read_or_make_transfer_table(TRANSFER_FUNCTION_PATH);
+
+for grid in table.grids
+    replace!(grid.lower_f, NaN => 0.0)
+    replace!(grid.upper_f, NaN => 0.0)
+end
+
+# emissivity profile table
+a_range = collect(range(0.0, 0.998, 10))
+h_range = collect(range(1.0, 10.0, 10))
+η_range = collect(range(0.01, 0.3, 10))
+
+prof_wrap = if isfile(LAMP_POST_PROFILE_PATH)
+    f = jldopen(LAMP_POST_PROFILE_PATH, "r")
+    prof = f["lamp_post"]
+    close(f)
+    prof
+else
+    prof_wrap = @time tabulate_emissivity_profile(a_range, h_range, η_range)
+    jldopen(LAMP_POST_PROFILE_PATH, "w"; compress=true) do f
+        f["lamp_post"] = prof_wrap
+    end
+end
+
+# Set up model
+# Note there are other options - see PR #3
+lp_model = LampPostThickDisc(
+    prof_wrap,
+    table;
+    K=FitParam(1e-4),
+    θ=FitParam(30.0, lower_limit=5, upper_limit=85),
+    E₀=FitParam(6.35),
+    a=FitParam(0.998, lower_limit=0.0, upper_limit=0.998),
+    h=FitParam(2.25, lower_limit=1.0, upper_limit=10.0),
+    η=FitParam(0.15),
+    # speed up model evaluation about 3 times
+    quadrature_points=13,
+    n_radii=600,
+)
+
+model = DeltaLine(K=FitParam(1e-5), E=FitParam(6.35, frozen=true)) + PowerLaw(K=FitParam(1e-2)) + AutoCache(lp_model)
```

Comment:
I don't think you need this DeltaLine component, since it's at the same energy as the LampPostThickDisc component? Removing it recovers the ability to remove the spin component in the result. I think this could be a bug in the DeltaLine component?